### ISSUE TITLE
negative updated results

### ISSUE THREAD

**Issue by florakarniav** [yahoo/streaming-benchmarks#36]
Hello, I've been using the yahoo streaming benchmark in different environments and since yesterday I tried it on an 8-core intel xeon machine. I see negative results in the updated file (e.g.  -70646). Can I get some help with this issue?

**Comment by Yitian-Zhang** [comment]
**This problem is most likely caused by the non-synchronization of the system time among worker nodes in your cluster.** When the benchmark system runs, **the start timestamp** will be created based on the system time of one node in the cluster, and then **the last updated timestamp** will be created based on other worker node (if the tasks were distributed among different nodes) when the final result written to Redis. Hence, if the time among different nodes is not synchronized, it is possible that the difference between the two timestamps is negative. **So you can check the system time of worker nodes in your cluster and then run the benchmark after keeping them synchronization.**

**Comment by varuy322** [comment]
> **This problem is most likely caused by the non-synchronization of the system time among worker nodes in your cluster.** When the benchmark system runs, **the start timestamp** will be created based on the system time of one node in the cluster, and then **the last updated timestamp** will be created based on other worker node (if the tasks were distributed among different nodes) when the final result written to Redis. Hence, if the time among different nodes is not synchronized, it is possible that the difference between the two timestamps is negative. **So you can check the system time of worker nodes in your cluster and then run the benchmark after keeping them synchronization.**

That's odd, I came across this issue. I checked the component configuration and code, nothing seemed to be wrong. Do you solve this by synchronize cluster system time?

**Comment by Yitian-Zhang** [comment]
> > **This problem is most likely caused by the non-synchronization of the system time among worker nodes in your cluster.** When the benchmark system runs, **the start timestamp** will be created based on the system time of one node in the cluster, and then **the last updated timestamp** will be created based on other worker node (if the tasks were distributed among different nodes) when the final result written to Redis. Hence, if the time among different nodes is not synchronized, it is possible that the difference between the two timestamps is negative. **So you can check the system time of worker nodes in your cluster and then run the benchmark after keeping them synchronization.**
> 
> That's odd, I came across this issue. I checked the component configuration and code, nothing seemed to be wrong. Do you solve this by synchronize cluster system time?

Yes, I encountered the same issue that negative updated values. And I solved this problem by synchronize cluster system time.

**Comment by HQBIAO** [comment]
> I came across this issue

Have you solved this problem,I came across this issue too
